docs-chef-io/content/supermarket/supermarket_upgrade.md

```diff
@@ -13,93 +13,194 @@ publishDate = 2022-01-03
     weight = 25
 +++
 
-
 ## Upgrade Matrix
 
-Running Version| Upgrade Version | Supported Version
----------|----------|---------
- A1 | B1 | C1
- A2 | B2 | C2
- A3 | B3 | C3
+  If running Supermarket 4.2, you can upgrade directly to the latest releases of Supermarket 4.3. If you are running a release with version less than 4.2 you must perform a stepped upgrade as outlined below.
```

Comment:
What prevents us from doing the upgrade from any version?

Reply:
@tas50 actually we need to keep track of what is the version of current PG for the upgrade. So we have added a software definition of pg-9.3 for handling the data migration through pg upgrade script. So only the supermarket versions that have pg-9.3 running we will be able to perform the upgrade directly.

Reply:
So we don't actually need 4.2? We just need a Supermarket release that has PG 9.3 right? That goes back quite far.

Reply:
Yes that's true. So in terms of PG dependency that is the case. Not sure if there would be any other blocker in case it's a major supermarket version upgrade.